File: Backend/StudentCareSystem.Domain/Helpers/Pagination.cs
using StudentCareSystem.Domain.Entities;

namespace StudentCareSystem.Domain.Helpers;

public class Pagination<T>
{
    public int TotalItems { get; set; }
    public int PageSize { get; set; }

    public int TotalPages
    {
        get
        {
            if (PageSize == 0) return 0;
            var temp = TotalItems / PageSize;
            return TotalItems % PageSize == 0 ? temp : temp + 1;
        }
    }

    public int PageIndex { get; set; }

    //page number start to 1
    public bool HasPreviousPage => PageIndex > 1;
    public bool HasNextPage => PageIndex < TotalPages;
    public ICollection<T> Items { get; set; } = [];

    public Pagination()
    {
        TotalItems = 0;
        PageSize = 1;
        PageIndex = 1;
    }
}

File: ai_service_agent/app/core/dtos/Pagination.py
from typing import Generic, List, TypeVar
from pydantic import BaseModel, Field

T = TypeVar('T')

class Pagination(BaseModel, Generic[T]):
    page_index: int = Field(..., ge=1, description="Current page number")
    page_size: int = Field(..., ge=1, description="Number of items per page")
    total_items: int = Field(..., ge=0, description="Total number of items")
    items: List[T] = Field(..., description="List of items on the current page")
    
    @property
    def total_pages(self) -> int:
        if self.page_size == 0:
            return 0
        temp = self.total_items // self.page_size
        return temp if self.total_items % self.page_size == 0 else temp + 1
    
    @property
    def has_previous_page(self) -> bool:
        """Check if there's a previous page available."""
        return self.page_index > 1

    @property
    def has_next_page(self) -> bool:
        """Check if there's a next page available."""
        return self.page_index < self.total_pages